pkg/apis/contrail/v1alpha1/zookeeper_types.go

```diff
@@ -82,88 +79,59 @@ func init() {
 }
 
 // InstanceConfiguration creates the zookeeper instance configuration.
-func (c *Zookeeper) InstanceConfiguration(request reconcile.Request,
+func (c *Zookeeper) InstanceConfiguration(request reconcile.Request, confCMName string,
 	podList *corev1.PodList,
 	client client.Client) error {
-	instanceConfigMapName := request.Name + "-" + "zookeeper" + "-configmap"
-	configMapInstanceDynamicConfig := &corev1.ConfigMap{}
-	err := client.Get(context.TODO(),
-		types.NamespacedName{Name: instanceConfigMapName, Namespace: request.Namespace},
-		configMapInstanceDynamicConfig)
+	zookeeperConfigInterface := c.ConfigurationParameters()
+	zookeeperConfig := zookeeperConfigInterface.(ZookeeperConfiguration)
+	pods := make([]corev1.Pod, len(podList.Items))
+	copy(pods, podList.Items)
+	sort.SliceStable(pods, func(i, j int) bool { return pods[i].Name < pods[j].Name })
+
+	confCMData, err := configtemplates.DynamicZookeeperConfig(podList, strconv.Itoa(*zookeeperConfig.ElectionPort), strconv.Itoa(*zookeeperConfig.ServerPort))
 	if err != nil {
 		return err
 	}
-	configMapInstancConfig := &corev1.ConfigMap{}
-	err = client.Get(context.TODO(),
-		types.NamespacedName{Name: instanceConfigMapName + "-1", Namespace: request.Namespace},
-		configMapInstancConfig)
+
+	var zookeeperLogBuffer, zookeeperXslBuffer bytes.Buffer
+	err = configtemplates.ZookeeperLogConfig.Execute(&zookeeperLogBuffer, struct{}{})
 	if err != nil {
```

Comment:
Nit: We can combine this err check with previous statement.

Reply:
it's not my code, i'm removing it in the next PR